### PR TITLE
SQS Plugin enhancements

### PR DESCRIPTION
**Pull request by mkozell**
Option to not delete messages after retrieval
Option to skip polling when the build queue reaches a defined size
Skip polling when Jenkins is scheduled for shutdown
Support for string and number message attributes

### CODE REVIEW

File: src/main/java/io/relution/jenkins/awssqs/SQSTriggerQueue.java [hunk 1]
```diff
@@ -251,6 +282,14 @@ public int getMaxNumberOfMessages() {
         return this.maxNumberOfMessages;
     }
 
+    @Override
+    public int getMaxNumberOfJobQueue() {
+        if (this.maxNumberOfJobQueue == null) {
+            return MAX_NUMBER_OF_JOB_QUEUE_DEFAULT;
+        }
+        return this.maxNumberOfJobQueue;
+    }
```
Comment (abeeskau) [inline src/main/java/io/relution/jenkins/awssqs/SQSTriggerQueue.java]: I prefer methods with a single return.  Default a local method variable to MAX_NUMBER_OF_JOB_QUEUE_DEFAULT.
Just my preference though.  It's up to you.

File: src/main/java/io/relution/jenkins/awssqs/threading/SQSQueueMonitorImpl.java [hunk 2]
```diff
@@ -169,14 +184,28 @@ private void execute() {
         }
     }
 
+    private boolean isQuietingDown() {
+        if (jenkins == null) {
```
Comment (abeeskau) [inline src/main/java/io/relution/jenkins/awssqs/threading/SQSQueueMonitorImpl.java]: Perhaps a single line would do.  Something like:
`return (jenkins ? jenkins.isQuietingDown() : false);`

File: src/main/java/io/relution/jenkins/awssqs/threading/SQSQueueMonitorImpl.java [hunk 3]
```diff
@@ -169,14 +184,28 @@ private void execute() {
...
+            return false;
+        }
+        return jenkins.isQuietingDown();
+    }
+
+    private int countBuildableItems() {
+        if (jenkins == null) { 
+            return 0;
+        }
+        return jenkins.getQueue().countBuildableItems();
```
Comment (abeeskau) [inline src/main/java/io/relution/jenkins/awssqs/threading/SQSQueueMonitorImpl.java]: Single line should do:
`return (jenkins ? jenkins.getQueue().countBuildableItems() : 0);`

**Comment by roribio** [comment]
Good suggestions @abeeskau 👍

File: src/main/java/io/relution/jenkins/awssqs/threading/SQSQueueMonitorImpl.java (hunk 2 above)
Comment (mkozell) [reply src/main/java/io/relution/jenkins/awssqs/threading/SQSQueueMonitorImpl.java]: Good idea. I switched to the single line version.

File: src/main/java/io/relution/jenkins/awssqs/SQSTriggerQueue.java (hunk 1 above)
Comment (mkozell) [reply src/main/java/io/relution/jenkins/awssqs/SQSTriggerQueue.java]: I switched to a single line version to avoid two returns

**Pull request merged**